src/companion.ts

```diff
@@ -149,7 +149,9 @@ const handleNewProblem = async (problem: Problem) => {
     let extn: string;
 
     if (defaultLanguage == null) {
-        const choices = Object.keys(config.extensions);
+        const allChoices = new Set(Object.keys(config.extensions));
+        const userChoices = getMenuChoices();
+        const choices = userChoices.filter((x) => allChoices.has(x));
```

Comment:
Here, it takes all the choices specified under `menuChoices` setting and filters out the ones supported by the extension so that illegal values are not shown in menu.